File: 78-subsets/78-subsets.cpp
class Solution {
public:
    vector<vector<int>> subs;
    
    void func(int i, int n, vector<int> &nums, vector<int> &sub){
        if(i>n-1){
            subs.push_back(sub);
            return;
        }    
        sub.push_back(nums[i]);
        func(i+1, n, nums, sub);
        sub.pop_back();
        func(i+1, n, nums, sub);
    }
    
    vector<vector<int>> subsets(vector<int>& nums) {
        int n = nums.size();
        vector<int> sub;
        func(0, n, nums, sub);
        return subs;
    }
};

File: 78-subsets/78-subsets.py
class Solution:
    def func(self, i, n, nums, subset, subs):
        if i > n-1:
            subs.append(subset[:])
            return
        
        subset.append(nums[i])
        self.func(i+1, n, nums, subset, subs)
        subset.pop()
        self.func(i+1, n, nums, subset, subs)
    
    
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        subs = []
        subset = []
        self.func(0, n, nums, subset, subs)
        return subs